buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/LicenseHeadersTask.groovy

```diff
@@ -33,8 +35,27 @@ import java.nio.file.Files
  */
 public class LicenseHeadersTask extends AntTask {
 
+    @OutputFile
+    File reportFile = new File(project.buildDir, 'reports/licenseHeaders/rat.log')
+
+    private List<FileCollection> javaFiles
+
     LicenseHeadersTask() {
         description = "Checks sources for missing, incorrect, or unacceptable license headers"
+        // Delay resolving the dependencies until after evaluation so we pick up generated sources
+        project.afterEvaluate {
+            List<FileCollection> javaFiles = project.sourceSets.collect({it.allJava})
+            setJavaFiles(javaFiles)
```

Comment:
I did! It doesn't work. It led me done a fun groovy rabbit hole. I think it doesn't work because of the decorated delegates and that groovy has a hack that'd make it work but they get in the way.